UnitTests/View/KeyboardEventTests.cs

```diff
@@ -0,0 +1,418 @@
+				return true;
+			}
+
+			OnInvokingKeyBindingsContinued = true;
+			return CancelVirtualMethods;
+		}
+
+		public override bool OnKeyPressed (KeyEventArgs keyEvent)
+		{
+			if (base.OnKeyPressed (keyEvent)) {
+				return true;
+			}
+
+			OnKeyPressedContinued = true;
+			return CancelVirtualMethods;
+		}
+
+		public override bool OnKeyUp (KeyEventArgs keyEvent)
+		{
+			if (base.OnKeyUp (keyEvent)) {
+				return true;
+			}
+
+			OnKeyUpContinued = true;
+			return CancelVirtualMethods;
+		}
+	}
+
+	[Theory]
+	[InlineData (true, false, false)]
+	[InlineData (true, true, false)]
+	[InlineData (true, true, true)]
+	public void Events_Are_Called_With_Only_Key_Modifiers (bool shift, bool alt, bool control)
+	{
+		bool keyDown = false;
+		bool keyPressed = false;
+		bool keyUp = false;
+
+		var view = new OnKeyTestView ();
+		view.CancelVirtualMethods = false;
+
+		view.KeyDown += (s, e) => {
+			Assert.Equal (KeyCode.Null, e.KeyCode & ~KeyCode.CtrlMask & ~KeyCode.AltMask & ~KeyCode.ShiftMask);
+			Assert.Equal (shift, e.IsShift);
+			Assert.Equal (alt, e.IsAlt);
+			Assert.Equal (control, e.IsCtrl);
+			Assert.False (keyDown);
+			Assert.False (view.OnKeyDownContinued);
+			keyDown = true;
+		};
+		view.KeyPressed += (s, e) => {
+			keyPressed = true;
+		};
+		view.KeyUp += (s, e) => {
+			Assert.Equal (KeyCode.Null, e.KeyCode & ~KeyCode.CtrlMask & ~KeyCode.AltMask & ~KeyCode.ShiftMask);
+			Assert.Equal (shift, e.IsShift);
+			Assert.Equal (alt, e.IsAlt);
+			Assert.Equal (control, e.IsCtrl);
+			Assert.False (keyUp);
+			Assert.False (view.OnKeyUpContinued);
+			keyUp = true;
+		};
+
+		//view.ProcessKeyDownEvent (new (Key.Null | (shift ? Key.ShiftMask : 0) | (alt ? Key.AltMask : 0) | (control ? Key.CtrlMask : 0)));
+		//Assert.True (keyDown);
+		//Assert.True (view.OnKeyDownWasCalled);
+		//Assert.True (view.OnProcessKeyDownWasCalled);
+
+		view.ProcessKeyDown (new KeyEventArgs (KeyCode.Null | (shift ? KeyCode.ShiftMask : 0) | (alt ? KeyCode.AltMask : 0) | (control ? KeyCode.CtrlMask : 0)));
+		Assert.True (keyPressed);
+		Assert.True (view.OnKeyDownContinued);
+		Assert.True (view.OnKeyPressedContinued);
+
+		view.ProcessKeyUp (new KeyEventArgs (KeyCode.Null | (shift ? KeyCode.ShiftMask : 0) | (alt ? KeyCode.AltMask : 0) | (control ? KeyCode.CtrlMask : 0)));
+		Assert.True (keyUp);
+		Assert.True (view.OnKeyUpContinued);
+	}
+
+	//[Fact]
+	//public void AllViews_OnKeyPressed_CallsResponder ()
+	//{
+	//	foreach (var view in TestHelpers.GetAllViews ()) {
+	//		if (view == null) {
+	//			_output.WriteLine ($"ERROR: null view from {nameof (TestHelpers.GetAllViews)}");
+	//			continue;
+	//		}
+	//		_output.WriteLine($"Testing {view.GetType().Name}");
+	//		var keyPressed = false;
+	//		view.KeyPressed += (s, a) => {
+	//			a.Handled = true;
+	//			keyPressed = true;
+	//		};
+
+	//		var handled = view.OnKeyPressed (new KeyEventArgs (Key.A));
+	//		Assert.True (handled);
+	//		Assert.True (keyPressed);
+	//		view.Dispose ();
+	//	}
+	//}
```

Comment:
Here is the test commented that prove the users don't have priority to cancel events. You commented without investigate what was wrong.